### ISSUE TITLE
Decomposing/Aggregating Cluster Jobs and Tool Evaluations

### ISSUE THREAD

**Issue by jmchilton** [galaxyproject/galaxy#4923]
Currently one tool evaluation corresponds to a single cluster job - this is somewhat limiting at the edges.

A cluster job can be a very heavy process - especially if processing large collections of items that should be processable quickly. It'd be nice if we had a way to link a bunch of collection jobs (or tool multi-run jobs) together with a single job that produces all the outputs.

This is the inverse of this is also the case - it'd be nice to separate out a single ``command`` section into multiple cluster jobs with different resource allocations (e.g. a single core for indexing, followed by many cores for mapping on that index, and possibly metadata in its own job).

Depending on the implementation - these tasks may be related or may be pursued in very different ways.

**Comment by dannon** [comment]
Sounds like we could build off (or completely replace) the parallelization work from years back with this?

**Comment by nsoranzo** [comment]
Yes, please! I was thinking about something similar some time ago for our GeneSeqToFamily workflow, where we have collections with thousand of small datasets processed in parallel. Each job is usually fast and most of the time is spent inside Galaxy for queueing/submitting/setting metadata.

**Comment by mvdbeek** [comment]
That would be fantastic in so many ways!
